backtrace-library/src/main/java/backtraceio/library/breadcrumbs/BacktraceBreadcrumbsLogger.java

```diff
@@ -0,0 +1,86 @@
+package backtraceio.library.breadcrumbs;
+
+import com.squareup.tape.QueueFile;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+import backtraceio.library.logger.BacktraceLogger;
+import backtraceio.library.enums.BacktraceBreadcrumbLevel;
+import backtraceio.library.enums.BacktraceBreadcrumbType;
+
+public class BacktraceBreadcrumbsLogger {
+
+    private final String LOG_TAG = BacktraceBreadcrumbsLogger.class.getSimpleName();
+
+    private long breadcrumbId = 0;
+
+    private BacktraceQueueFileHelper backtraceQueueFileHelper;
+
+    /**
+     * We truncate messages longer than this
+     */
+    private final int maxMessageSizeBytes = 1024;
+
+    /**
+     * We truncate serialized attribute strings longer than this
+     */
+    private final int maxSerializedAttributeSizeBytes = 1024;
+
+    public BacktraceBreadcrumbsLogger(String breadcrumbLogDirectory, int maxQueueFileSizeBytes) throws IOException, NoSuchMethodException {
+        this.backtraceQueueFileHelper = new BacktraceQueueFileHelper(breadcrumbLogDirectory, maxQueueFileSizeBytes);
+    }
+
+    public boolean addBreadcrumb(String message, Map<String, Object> attributes, BacktraceBreadcrumbType type, BacktraceBreadcrumbLevel level) {
+        // We use currentTimeMillis in the BacktraceReport too, so for consistency
+        // we will use it here.
+        long time = System.currentTimeMillis();
+
+        message = StringUtils.abbreviate(message, maxMessageSizeBytes);
```

Comment:
Are we import whole library `implementation 'org.apache.commons:commons-lang3:3.0'` just to use it once here for something what could be done by `(substring(str, 0, max-3) + "...").` ?

Reply:
That's a good point, I will avoid this import and use included string libraries for this purpose. It definitely doesn't seem a good enough reason to include this whole library.